### ISSUE TITLE
Completions Install not working

### ISSUE THREAD

**Issue by gjrtimmer** [DannyBen/bashly#448]
### Bashly Version

Latest Ruby Gem

### Description

Trying to install the completions on my Mac, I'm getting the following error:

```shell
➜  ~ bashly completions -i    
 NameError 
uninitialized constant Bashly::Commands::Completions::Completely
```

What is essential to know, when trying to install bashly with `brew gem install bashly` the install complained about needing ruby 2.7 and 2.6 was installed.

ERROR:
```shell
➜  ~ brew gem install bashly
==> Fetching dependencies for gem-bashly: ruby
==> Fetching ruby
==> Downloading https://ghcr.io/v2/homebrew/core/ruby/manifests/3.2.2_1
################################################################################################################# 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/ruby/blobs/sha256:6a1a2f3c6404c532cce812679b55469831214ed526344cdef8913
################################################################################################################# 100.0%
==> Fetching gem-bashly
==> Fetching bashly from gem source
Fetching bashly-1.1.2.gem
Downloaded bashly-1.1.2
Warning: Cannot verify integrity of 'bashly-1.1.2.gem'.
No checksum was provided.
For your reference, the checksum is:
  sha256 "33918d797ba23f82adad9410462c3cf145c30c205cd9a5e62e4d1415d2a089d7"
==> Installing dependencies for gem-bashly: ruby
==> Installing gem-bashly dependency: ruby
==> Downloading https://ghcr.io/v2/homebrew/core/ruby/manifests/3.2.2_1
Already downloaded: /Users/gtimmer/Library/Caches/Homebrew/downloads/b7df479895630ebb7fdc5e0915854094e1f610797d6eff15202cebf668cbad53--ruby-3.2.2_1.bottle_manifest.json
==> Pouring ruby--3.2.2_1.arm64_sonoma.bottle.tar.gz
🍺  /opt/homebrew/Cellar/ruby/3.2.2_1: 16,605 files, 48.3MB
==> Installing gem-bashly
==> /usr/bin/gem install /Users/gtimmer/Library/Caches/Homebrew/bashly-1.1.2.gem --no-document --no-wrapper --no-user-in
Last 15 lines from /Users/gtimmer/Library/Logs/Homebrew/gem-bashly/01.gem:
2023-11-17 06:19:20 +0000

/usr/bin/gem
install
/Users/gtimmer/Library/Caches/Homebrew/bashly-1.1.2.gem
--no-document
--no-wrapper
--no-user-install
--install-dir
/opt/homebrew/Cellar/gem-bashly/1.1.2
--bindir
/opt/homebrew/Cellar/gem-bashly/1.1.2/bin

ERROR:  Error installing /Users/gtimmer/Library/Caches/Homebrew/bashly-1.1.2.gem:
	stringio requires Ruby version >= 2.7. The current ruby version is 2.6.10.210.

Do not report this issue to Homebrew/brew or Homebrew/homebrew-core!

```

I moved past this by added `--homebrew-ruby` to the install process, maybe this is linked to why the complations will not install?

### Reproduction Steps

Install lastest version of bashly on MacOS

### Expected Behavior

To install the completions

**Comment by DannyBen** [comment]
Thanks for reporting.

This problem seems to occur regardless of operating system. I fixed it in #449.

I also noticed that the installation via homebrew is not reliable. I will remove it from the documentation site.

To install on mac, it seems better to simply install ruby, and then use the standard installation instructions.
On mac, bash is also outdated, so installing an up to date bash is needed.

```
$ brew install bash ruby
$ gem install bashly
$ bashly
```

There is a debug action output [here](https://github.com/DannyBen/bashly/actions/runs/6901147663/job/18775448333) if you want to see the run on mac.

---

If you want to install these completions before I release the new version (which will happen in a few days) - you can simply copy [this completions file](https://github.com/DannyBen/bashly/blob/master/lib/bashly/completions/bashly-completions.bash) manually to your completions directory, and name the file `bashly`

I believe on mac it should be named `/usr/share/bash-completion/completions/bashly`

**Comment by gjrtimmer** [comment]
Thank you, I did update bash btw

**Comment by gjrtimmer** [comment]
What will be the upgrade command: `gem upgrade bashly`?

**Comment by gjrtimmer** [comment]
When I try to install with the above commands I run into the following error:

```shell
ERROR:  Error installing bashly:
	The last version of rouge (>= 3.14, < 5.0) to support your Ruby & RubyGems was 3.30.0. Try installing it with `gem install rouge -v 3.30.0` and then running the current command again
	rouge requires Ruby version >= 2.7. The current ruby version is 2.6.10.210.
```

**Comment by DannyBen** [comment]
As can be seen in the [debug action](https://github.com/DannyBen/bashly/actions/runs/6901147663/job/18775448333), the installation on mac should be very straight forward. You should not need to install `rouge` or anything else other than `bashly` itself.

I am not using mac, but my guess is that something with the ruby (or old bashly) installation is problematic in your system.

So - I would do this:

1. Undo the brea bashly gem installation - I believe it would be `brew gem uninstall bashly`
2. Uninstall the `brew-gem` brew package - `brew uninstall brew-gem`
3. Install ruby and bash - `brew install bash ruby`
4. Make sure your ruby is at least 3.0 - `ruby -v`
5. Install the gem normally - `gem install bashly`

**Comment by DannyBen** [comment]
> What will be the upgrade command: `gem upgrade bashly`?

Same as other package managers - just use `install`, it will upgrade if needed. So `gem install bashly`.

**Comment by gjrtimmer** [comment]
I think I found the culprit, its the Ruby shipped with MacOS.

For MacOS you need to make changes to your shell: `~/.zshrc` to load the brew installed version with higher priority in your path. Maybe worthy to put in your documentation.

## Intel
```shell
if [ -d "/usr/local/opt/ruby/bin" ]; then
  export PATH=/usr/local/opt/ruby/bin:$PATH
  export PATH=`gem environment gemdir`/bin:$PATH
fi
```

## Silicon
```shell
if [ -d "/opt/homebrew/opt/ruby/bin" ]; then
  export PATH=/opt/homebrew/opt/ruby/bin:$PATH
  export PATH=`gem environment gemdir`/bin:$PATH
fi
```

Then it will finally show ruby as version 3.x

**Comment by gjrtimmer** [comment]
Yeah, after making this change to my `.zshrc` it finally all worked, its the native ruby on MacOS that is in the way. Hope this is a great addition to your documentation.

**Comment by DannyBen** [comment]
Well

1. in my tests on GitHub Actions mac, there was no need for such a change
2. Installing Ruby or debugging Homebrew is beyond the scope of bashly's documentation - although I do provide some guidance [here](https://bashly.dannyb.co/installing-ruby/).

**Comment by gjrtimmer** [comment]
I have a very recent Mac so something may have changed.

The following should work for MacOS.

-- START DOCS --

Install `bashly` on MacOS with homebrew

## Install Ruby

```shell
brew install ruby
```

Update `~/.zshrc` to load ruby from brew with higher priority then native MacOS brew

### Intel
```shell
if [ -d "/usr/local/opt/ruby/bin" ]; then
  export PATH=/usr/local/opt/ruby/bin:$PATH
  export PATH=`gem environment gemdir`/bin:$PATH
fi
```

### Silicon
```shell
if [ -d "/opt/homebrew/opt/ruby/bin" ]; then
  export PATH=/opt/homebrew/opt/ruby/bin:$PATH
  export PATH=`gem environment gemdir`/bin:$PATH
fi
```

## Install `bashly`

`gem install bashly`

-- END DOCS --

**Comment by gjrtimmer** [comment]
Just verified everything on MacBook Pro M2.

**Comment by DannyBen** [comment]
> The following should work for MacOS.

Installing ruby with homebrew should just be `brew install ruby` - if you are required to update anything manually, someone need to change their install process - either The [Homebrew formula maintainer](https://formulae.brew.sh/formula/ruby), or the zsh maintainers.

**Comment by DannyBen** [comment]
Fixed in version 1.1.3
